feat: Use Anthropic Marketplace JSON in Marketplace Repo #438 (#454)

adriander opened the pull request:
This restores the changes from #445.

- Added Anthropic Marketplace JSON type definitions
- Implemented marketplace manifest fetche
- Updated marketplace services to fetch from JSON
- Updated install/uninstall commands to use async marketplace service lookup
- Added comprehensive test coverage for all new functionality

## Checklist

- [x] Follow the [contributor guide](./CONTRIBUTING.md)
- [x] End-to-end tests pass
- [x] Unit tests for essential parts of your code, e.g. APIs exposed via SDKs or endpoints
- [x] Update `./docs`
- [x] Recognize contributors with "@all-contributors please add <person> for <code, bug, docs, etc>"
- [x] Linked issues #438

iliassjabali committed:
feat: Use Anthropic Marketplace JSON in Marketplace Repo #438 (#454)

This restores the changes from #445.

- Added Anthropic Marketplace JSON type definitions
- Implemented marketplace manifest fetcher with caching
- Updated marketplace services to fetch from JSON with fallback
- Added --refresh flag to marketplace list command
- Updated install/uninstall commands to use async marketplace service
lookup
- Added comprehensive test coverage for all new functionality

## Checklist

- [x] Follow the [contributor guide](./CONTRIBUTING.md)
- [x] End-to-end tests pass
- [x] Unit tests for essential parts of your code, e.g. APIs exposed via
SDKs or endpoints
- [x] Update `./docs`
- [x] Recognize contributors with "@all-contributors please add <person>
for <code, bug, docs, etc>"
- [x] Linked issues #438

havasik committed:
Merge branch 'main' into feat/migrate-ark-marketplace-following-anthropic-plugin-marketplaces

havasik committed:
Merge branch 'main' into feat/migrate-ark-marketplace-following-anthropic-plugin-marketplaces

havasik committed:
feat: remove broken refresh / cache mechanism

havasik committed:
feat: remove fallback marketplace services and caching

Remove hardcoded fallback services and caching mechanism for marketplace.json. The CLI now fetches services directly from the marketplace repository and displays a clear unavailable message when the marketplace cannot be reached.

Changes:
- Remove hardcoded fallback services (phoenix, langfuse, noah)
- Remove caching logic from marketplaceFetcher and marketplaceServices
- Remove --refresh flag from marketplace list command
- Update install/uninstall commands to handle unavailable marketplace
- Show clear "Marketplace unavailable" message when fetch fails
- Update all related tests

havasik committed:
feat: add marketplace agents support with type-based filtering

Separate marketplace items into services and agents based on type field.
Services use marketplace/services/* path, agents use marketplace/agents/*.
Update CLI commands and tests to support both categories.

havasik committed:
Merge branch 'main' into feat/migrate-ark-marketplace-following-anthropic-plugin-marketplaces

havasik committed:
fix: remove duplicate Agents section in marketplace list command

havasik committed:
docs: update marketplace documentation to remove caching references

dwmkerr commented on tools/ark-cli/src/lib/marketplaceFetcher.ts:
```diff
@@ -0,0 +1,104 @@
+  AnthropicMarketplaceItem,
+} from '../types/marketplace.js';
+
+const MARKETPLACE_REPO_URL =
```
could go in config defaults then able to be overriden by `.arkrc.yaml` but that could be a follow on

dwmkerr reviewed (approved):
approved but consider config

havasik committed:
feat: make marketplace URLs configurable via .arkrc.yaml

Move marketplace repository URL and registry from hardcoded constants
to configuration system with full .arkrc.yaml support.

Configuration hierarchy:
1. Defaults in loadConfig()
2. User config from ~/.arkrc.yaml
3. Project config from .arkrc.yaml
4. Environment variables (ARK_MARKETPLACE_REPO_URL, ARK_MARKETPLACE_REGISTRY)

Users can now override marketplace locations:
```yaml
marketplace:
  repoUrl: https://example.com/my-marketplace
  registry: oci://example.com/charts
```

havasik committed:
feat: unify marketplace URL configuration into single source of truth

Move all hardcoded marketplace URLs to use helper functions that load
from config. This eliminates duplication across multiple files.

Changes:
- Add getMarketplaceRepoUrl() and getMarketplaceRegistry() helpers in config.ts
- Update marketplaceFetcher.ts to use helper functions
- Update arkServices.ts to use getMarketplaceRegistry()
- Update marketplace command to use helper functions for display
- Update all tests to mock helper functions with test values
- Remove redundant fallback values from helper functions

All marketplace URLs now defined once in config.ts defaults with full
.arkrc.yaml and environment variable override support.
